File: ChessGameRCdb/ClientAppBoard/src/board/interface/IMove.ts
import IFigure from 'board/interface/IFigure'
import ISquare from 'board/interface/ISquare'
import { FigureAction } from 'board/enum/FigureAction'

export default interface IMove {
    Figure: IFigure
    Square : ISquare
    Type: FigureAction
    IsExecutable: boolean
}

File: ChessGameRCdb/Logic/Interfaces/IMove.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ChessGame.Logic
{
    public interface IMove
    {
        //MoveType MoveType { get; set; }
        IEnumerable<MoveOption> GetMoveOptions(IBoard board, IFigure figure, Direction direction);
    }
}
